components/local-app/main.go

```diff
@@ -108,6 +108,14 @@ func main() {
 				},
 				Value: 30,
 			},
+			&cli.StringFlag{
+				Name:  "timeout",
+				Usage: "How long the local app can run if last workspace was stopped. like [3h] default is forever",
```

Comment:
It seems that this PR does something else: It times out the local app if last ssh connection was closed. I am not mind generally since it should be enough in case of VS Code support. 

So maybe rename it to `ssh-idle-timeout` and change description?

Reply:
Yes, I terminated the local-app with a timeout after the last ssh connect stopped. I did that because I think it has no necessary to keep the local-app while server still have workspace alive but no ssh connected to them.

Or may be I got a misunderstanding of what issue #7074 want to do?

Reply:
I mean an original idea was that timeout only stop if nothing is going on, i.e. no workspaces at all.
This PR though can timeout the app if there are running workspace and they even can have some connection tunnelled (not via SSH).

It is not bad in itself, since it is enough for VS Code, but then options should be changed. Current option does not work based on `How long the local app can run if last workspace was stopped.`.

Alternatively we should `timeout` flag to respect lifecycle of workspaces instead of ssh connection. Maybe it can be done by generally tracking ssh tunnels even if there are not connections? There should be one ssh tunnel per a workspace and when workspace times out in 30 mins when ssh tunnel should be removed.